presto-tests/src/test/java/io/prestosql/tests/TestTpchDistributedQueries.java

```diff
@@ -48,22 +46,7 @@ public void testIOExplain()
         MaterializedResult result = computeActual("EXPLAIN (TYPE IO, FORMAT JSON) " + query);
         IoPlanPrinter.IoPlan.TableColumnInfo input = new IoPlanPrinter.IoPlan.TableColumnInfo(
                 new CatalogSchemaTableName("tpch", "sf0.01", "orders"),
-                ImmutableSet.of(
```

Comment:
Good catch. As i understand, `applyFilter` was called only when there was some Filter. So this lack of "pull-up" is pre-existing. 
But the change here also means we could have just made it worse and it may bite in practice.

Reply:
> Hmm... this is potentially a problem. Technically, the connector should still be able to describe the constraints of a given table, regardless of whether a filter was "pushed in" or not. 

@martint this is already an issue. Please see: https://github.com/prestosql/presto/issues/1399. Currently detailed predicate (e.g. involving partitions) is not returned back to planner as `TableScanNode#enforcedConstraint` is derived from pushed down predicate.